[Documentation] Add dependency info to test docs

elliotolds opened the pull request:
When following the build instructions and then trying to run the RPC tests, 
a unix user would get an error when python tried to import zmq.

This change updates the docs to let the user know how to install
the required dependencies before running the RPC tests.

There may be other dependencies that should be added to the docs,
particularly ones for non-unix systems. This is the only unlisted
dependency I encountered using linux.

elliotolds commented:
I could also make the following change to the test itself, to check whether zmq is installed and warn the user if not: https://github.com/elliotolds/bitcoin/commit/ed067f722a24

However, I'm not planning to submit that pull request unless people request it, because I feel the benefit is minor and not worth the extra code complexity.

laanwj commented:
elliotolds@ed067f7 makes a lot of sense to me in combination with this documentation change.

jmcorgan commented:
ACK.  I agree also on ed067f7, except it has a typo ("import zmp") on line 49.

paveljanik commented on README.md:
```diff
@@ -55,7 +55,7 @@ submit new unit tests for old code. Unit tests can be compiled and run
+These tests can be run (if the [test dependencies](/qa) are installed) with: `qa/pull-tester/rpc-tests.py`
 
 The Travis CI system makes sure that every pull request is built for Windows
 and Linux, OSX, and that unit and sanity tests are automatically run.
```
Can you please also change OSX -> OS X when you are touching this file?

elliotolds replied:
Done.

paveljanik commented on qa/README.md:
```diff
@@ -5,6 +5,16 @@ Every pull request to the bitcoin repository is built and run through
+=================
+Before running the tests, the following must be installed.
+
+Unix
```
Looks like the command here doesn't even apply to all GNU/Linux based systems but only to apt-get based systems. Unix -> Linux at least?

elliotolds replied:
Just updated with a clarification that the instructions are for Ubuntu and Debian, as done in /doc/build-unix.md

MarcoFalke commented on qa/README.md:
````diff
@@ -5,6 +5,17 @@ Every pull request to the bitcoin repository is built and run through
+sudo apt-get install python-zmq
+```
+
 Running tests
````
Nit: As you are already touching this file, mind to add a line to mention an undocumented arg?

`--coverage` (this generates a basic coverage report for the RPC)

laanwj commented:
utACK after squash

elliotolds committed:
Check if zmq is installed in tests, update docs

If ZMQ is enabled, check whether it's installed before running ZMQ tests.
If it isn't, disable ZMQ and print a warning.

Also add dependency info to test docs, so users know ZMQ is required
before running tests, and so they know how to install it.

When following the build instructions before this change then trying
to run the RPC tests, a unix user would get an error when python
tried to import zmq.

There may be other dependencies that should be added to the docs,
particularly ones for non-unix systems. This is the only unlisted
dependency I encountered using linux.
